.github/workflows/chart.yaml

```diff
@@ -1,65 +1,67 @@
 name: Chart build
 
 on:
+  workflow_dispatch: {}
   pull_request:
     branches:
       - develop
 
 jobs:
-  build-with-cost-model-develop:
+  test-chart:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
 
-      - name: Set up Helm
-        uses: azure/setup-helm@v1
-        with:
-          version: v3.8.2
-
-      - name: Run helm lint
+      - name: Helm lint
         working-directory: ./cost-analyzer
         run: helm lint
 
-      - name: Run helm template
+      - name: Helm template
         working-directory: ./cost-analyzer
-        # Template the chart out for kubeval usage and also validates that it can 'build'
         run: helm template . --set global.prometheus.enabled=false --set global.grafana.enabled=false > full.yaml
 
-      - name: Kubeval
-        uses: instrumenta/kubeval-action@master
+      - name: Kubeconform
+        uses: docker://ghcr.io/yannh/kubeconform:latest
         with:
-          files: ./cost-analyzer/full.yaml
-          ignore_missing_schemas: false
+          entrypoint: /kubeconform
+          args: "-summary -output text ./cost-analyzer/full.yaml"
 
-      # Setup a short sha to use for cluster naming to ensure we create and cleanup unique clusters and set the k8s version for
-      # both kubectl and k3s
-      - name: Setup env
-        run: |
-          echo "short_sha=${GITHUB_SHA::10}" >> $GITHUB_ENV
-          echo "k8s_version=v1.23.7" >> $GITHUB_ENV
 
-      - uses: azure/setup-kubectl@v3.0
-        with:
-          version: ${{ env.k8s_version }}
+  # Installs the chart across a matrix of modern Kubernetes versions
+  deploy-chart:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s-version:
+          - name: v1.23
+            version: v1.23.17
+          - name: v1.24
+            version: v1.24.15
+          - name: v1.25
+            version: v1.25.11
+          - name: v1.26
+            version: v1.26.6
+          - name: v1.27
+            version: v1.27.3
+          - name: v1.28
+            version: v1.28.0
+    needs: test-chart
+    name: ${{ matrix.k8s-version.name }} test
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
 
-      - uses: AbsaOSS/k3d-action@v2
-        name: "Create Single Cluster"
+      - name: Create KinD cluster
```

Comment:
Any specific reason you chose to migrate from k3s to KinD?

Reply:
There is a curated action by the Helm program which simplifies workflows and allows us to simply pick up that logic without having to reinvent it. Plus, KinD is somewhat of the defacto standard because it's from a Kubernetes SIG. I personally love and use K3d as my system, but there are some quirks it introduces.